fdbclient/ReadYourWrites.actor.cpp

```diff
@@ -1604,7 +1605,9 @@ void ReadYourWritesTransaction::set( const KeyRef& key, const ValueRef& value )
 	if(key >= getMaxWriteKey())
 		throw key_outside_legal_range();
 
-	if(options.readYourWritesDisabled ) {
+	if (options.readYourWritesDisabled) {
+		approximateSize += key.expectedSize() + value.expectedSize() + sizeof(MutationRef) +
+		                   (addWriteConflict ? sizeof(KeyRangeRef) : 0);
```

Comment:
If we are accounting for a conflict range here, I think we may also need to add `2 * key.expectedSize() + 1` for the keys in the range.

Also, do we need to add the conflict range cost here regardless of whether we are bypassing RYW? It looks like the conflict ranges aren't added until commit time (or possibly during usage of versionstamps), so we would probably also need to add a parameter to the conflict range adding functions to avoid updating the approximate size when called from those contexts.

Reply:
For read conflict ranges, add in `ReadYourWritesTransaction::updateConflictMap` or direct add for `readYourWritesDisabled` case.

For write conflict ranges, don't add their size in the commit time by adding a flag to `ReadYourWritesTransaction::writeRangeToNativeTransaction`. The idea is that, after commit, client code should call `Transaction::getSize()` to get transaction size.